lib/services/sts.js

```diff
@@ -43,5 +46,87 @@ AWS.util.update(AWS.STS.prototype, {
 
   assumeRoleWithSAML: function assumeRoleWithSAML(params, callback) {
     return this.makeUnauthenticatedRequest('assumeRoleWithSAML', params, callback);
+  },
+
+  /**
+   * @api private
+   */
+  validateRegionalEndpointsFlag: function validateRegionalEndpointsFlag() {
+    //validate config value
+    var config = this.config;
+    if (
+      typeof config.stsRegionalEndpoints === 'string' &&
+      ['legacy', 'regional'].indexOf(config.stsRegionalEndpoints.toLowerCase()) >= 0
+    ) {
+      config.stsRegionalEndpoints = config.stsRegionalEndpoints.toLowerCase();
+      return;
+    }
+    if (config.stsRegionalEndpoints) {
+      throw AWS.util.error(new Error(), {
+        code: 'InvalidConfiguration',
+        message: 'invalid "stsRegionalEndpoints" configuration. Expect "legacy" ' +
+        ' or "regional". Got "' + config.stsRegionalEndpoints + '".'
+      });
+    }
+    if (!AWS.util.isNode()) return;
+    //validate environmental variable
+    if (Object.prototype.hasOwnProperty.call(process.env, ENV_REGIONAL_ENDPOINT_ENABLED)) {
+      var envFlag = process.env[ENV_REGIONAL_ENDPOINT_ENABLED];
+      if (typeof envFlag === 'string' && ['legacy', 'regional'].indexOf(envFlag.toLowerCase()) >= 0) {
```

Comment:
This if-else be moved to a different function, as it's similar across the three blocks